File: Lab5/src/KI306/Voitovych/Lab5/Writer.java
/**
 * Lab5
 */
package KI306.Voitovych.Lab5;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Scanner;

/**
 * Writer is class for write and read result in .txt or .bin file
 */
public class Writer {
	
	private double result;
	
	/**
	 * Constructor
	 */
	
	public Writer() {
		
	}
	
	/**
	 * Constructor
	 * @param <code>result</code> result
	 */
	
	public Writer(double result) {
		this.result = result;
	}
	
	/**
	 * Method sets for result
	 */
	
	public void setResult(double result) {
		this.result = result;
	}
	
	/**
	 * Method gets for result
	 */
	
	public double getResult() {
		return this.result;
	}
	/**
	 * Method for write result in text file
	 */
	
	public void wrTxt(String fName) throws FileNotFoundException{
		PrintWriter f = new PrintWriter(fName);
		f.printf("%f ",result);
		f.close();
	}
	/**
	 * Method for read result in text file
	 */
	public double rdTxt(String fName){
		try{
			File f = new File (fName);
			if (f.exists()){
				Scanner s = new Scanner(f);
				result = s.nextDouble();
				s.close();
				return result;
			}
			else throw new FileNotFoundException("File " + fName + "not found");
		}
		catch (FileNotFoundException ex){
			System.out.print(ex.getMessage());
		}
		return result;
	}
	/**
	 * Method for write result in bin file
	 */
	
	public void wrBin(String fName) throws FileNotFoundException, IOException{
		DataOutputStream f = new DataOutputStream(new FileOutputStream(fName));
		f.writeDouble(result);
		f.close();
	}
	/**
	 * Method for read result in bin file
	 */
	
	public double rdBin(String fName) throws FileNotFoundException, IOException{
		DataInputStream f = new DataInputStream(new FileInputStream(fName));
		result = f.readDouble();
		f.close();
		return result;
	}

}

File: Lab8/writer.py
import os
import struct

#Write in text file
def wr_txt(f_name, result):
    with open(f_name, 'w') as f:
        f.write(f"{result:.6f}\n")

#Read in text file
def rd_txt(f_name):
    try:
        if os.path.exists(f_name):
            with open(f_name, 'r') as f:
                result = float(f.readline())
            return result
    except FileNotFoundError as ex:
        return "File {f_name} not found"

#Write in bin file
def wr_bin(f_name, result):
    with open(f_name, 'wb') as f:
        f.write(struct.pack('d', result))

#Read in bin file
def rd_bin(f_name):
    try:
        with open(f_name, 'rb') as f:
            result = struct.unpack('d', f.read(8))[0]
        return result
    except FileNotFoundError as ex:
        print(ex)
        return "File {f_name} not found"
